samples/winapi_filefind/main.c

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include <fileapi.h>
+#include <hal/winerror.h>
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+
+int main()
+{
+    int ret = pb_init();
+    if (ret != 0) {
+        XSleep(2000);
+        return -1;
+    }
+
+    pb_show_debug_screen();
+
+    debugPrint("Content of C:\\\n");
+
+    WIN32_FIND_DATA findFileData;
+    HANDLE hFind;
+
+    hFind = FindFirstFile("C:\\*.*", &findFileData);
```

Comment:
Why is this `\\*.*` and not just `\\*`? Will this also find files without a dot in their name?

Reply:
`\\*.*` and `\\*` behave the same, both will find all files, independent of whether they contain a dot or not. My implementation of FindFirstFile() contains [code](https://github.com/XboxDev/nxdk/pull/112/commits/551a39ae4234d7cff5d937f14af17029f36ecda2#diff-71556cfa5b0777b50584eaf8d9d2223dR70) to explicitly handle the `\\*.*` special case.
My intention for using `\\*.*` was to demonstrate that it works just like it does on Windows (I recall seeing several samples for file finding using that).

Reply:
Interesting; You should add a comment here (about it not caring about the dot) and a comment that explains this "hack" in the linked mask-handler code.

- Did you verify this actually happens on Windows? `*.*` *really* listing files without dot?
- Are there other known differences to Windows / XDK FindFile you are aware of? What about `?` wildcard?

Reply:
I believe I did check it before, but I just ran a small sample in my Windows 10 VM to be sure, and I can confirm that `*.*` works as described and identical to `*`.
Tbh I wasn't aware of the `?` wildcard, but I'll try to write some more test code covering that later today.
I'll push some changes including the requested comments in a few minutes.

Reply:
I checked XAPI disassembly earlier, and the check you do is the only one I could spot.
They do a `memcmp` instead of `strncmp` though (so it won't check for a `\0`; it checks the length for 3, then assumes the string is 3 symbols that match exactly).

Reply:
I swapped `strncmp` for `memcmp` and wrote a bit of test code to check `?` wildcard behavior, and from what I can tell, it seems to work correctly: https://gist.github.com/thrimbor/34f1914f9c62f5b584bc382605d854a6